src/crimson/os/seastore/transaction_manager.cc

```diff
@@ -88,23 +88,49 @@ TransactionManager::mount_ertr::future<> TransactionManager::mount()
   cache->init();
   return epm->mount(
   ).safe_then([this] {
-    return journal->replay(
-      [this](
-	const auto &offsets,
-	const auto &e,
-	const journal_seq_t &dirty_tail,
-	const journal_seq_t &alloc_tail,
-	sea_time_point modify_time)
-      {
-	auto start_seq = offsets.write_result.start_seq;
-	return cache->replay_delta(
-	  start_seq,
-	  offsets.record_block_base,
-	  e,
-	  dirty_tail,
-	  alloc_tail,
-	  modify_time);
+    return seastar::do_with(
+    std::map<paddr_t, delta_info_t>(),
+    [this](auto &on_replayed_delta) {
+      return journal->replay(
+	[this, &on_replayed_delta](
+	  const auto &offsets,
+	  const auto &e,
+	  const journal_seq_t &dirty_tail,
+	  const journal_seq_t &alloc_tail,
+	  sea_time_point modify_time)
+	{
+	  auto start_seq = offsets.write_result.start_seq;
+	  return cache->replay_delta(
+	    start_seq,
+	    offsets.record_block_base,
+	    e,
+	    dirty_tail,
+	    alloc_tail,
+	    modify_time
+	  ).safe_then([&e, &on_replayed_delta](auto ret) {
+	    if (ret && e.is_inplace_rewrite()) {
+	      on_replayed_delta[e.paddr]= e;
+	    }
+	    return Cache::replay_delta_ertr::make_ready_future<bool>(ret); 
+	  });
+      }).safe_then([this, &on_replayed_delta] {
+	return crimson::do_for_each(on_replayed_delta,
+	 [this](auto &p) {
+	  return with_transaction_weak(
+	    "mount",
+	    [this, &p](auto &t)
+	  {
+	    return cache->get_extent_if_cached(t,
```

Comment:
Valid inplace-rewrite extents should be dirty extents which must be always present in Cache and alive in SeaStore. I think it should be better to somehow track paddr -> (extent, expected_crc) instead, and check the correctness of expected_crc at the end of RBM jounal replay.

Reply:
Please see the last commit.